src/components/TableOfContents/index.tsx

```diff
@@ -29,17 +33,24 @@ function ToCItem<T, K extends string | number>(props: ToCItemProps<T, K>) {
         onChange,
         keySelector,
         labelSelector,
+        idSelector,
         childrenSelector,
         defaultCollapseLevel,
     } = props;
 
-    const id = keySelector(option);
+    const key = keySelector(option);
+    const id = idSelector(option);
     const title = labelSelector(option);
     const children = childrenSelector(option);
 
     const handleClick = useCallback(
-        () => onChange(id),
-        [id, onChange],
+        () => {
+            if (isDefined(key)) {
+                return onChange({ key, id });
+            }
+            return undefined;
```

Comment:
Why are we returning? onChange has no return value.

Reply:
some of TOC elements don't have the corresponding key and aren't filterable. undefined is returned as to be consistent with eslint rules.

Reply:
The gist is that we don't need to return anything. Why return at all?